scompose/project/project.py

```diff
@@ -50,9 +50,11 @@ def get_instance_names(self):
            are defined.
         """
         names = []
-        if self.config is not None:
-            if "instances" in self.config:
-                names = list(self.config["instances"].keys())
+        if self.instances is not None:
+            names = self.instances.keys()
+        elif self.config is not None and "instances" in self.config:
+            names = list(self.config["instances"].keys())
+
```

Comment:
My thinking here is that we should not again be parsing the self.config - the parse() function called on init should already parse instances into self.instances. I don't see a case where we would want to derive names from self.config["instances"] because they should already be in self.instances. Thoughts?